app/questionnaire/questionnaire_store_updater.py

```diff
@@ -438,8 +438,9 @@ def started_section_keys(
     ) -> list[SectionKeyType]:
         return self._progress_store.started_section_keys(section_ids)
 
-    def dependency_has_dynamic_answers(self, block_id):
-        if question := self._schema.get_block(block_id).get("question"):
+    def dependency_has_dynamic_answers(self, block_id: str) -> bool:
```

Comment:
This would error for variant, but I still don't get why this is needed?. There is an outstanding comment for it.

Reply:
I think we don't need this now since we refactored dynamic answers resolution (and ids resolution till the very last step before form is displayed). I've tested it and works fine so will remove it now.